mllib/src/main/scala/org/apache/spark/mllib/linalg/distributed/RowMatrix.scala

```diff
@@ -212,7 +212,7 @@ class RowMatrix(
       tol: Double,
       mode: String): SingularValueDecomposition[RowMatrix, Matrix] = {
     val n = numCols().toInt
-    require(k > 0 && k <= n, s"Request up to n singular values but got k=$k and n=$n.")
+    require(k > 0 && k <= n, s"Request up to n singular values but got k=$k and numCols=$n.")
```

Comment:
Thanks, I got confused by the condition `k <= n` . Which made me read it as "The maximum singular values possible is n, but provided with k (which is greater than n)."